test/conformance/generatename_test.go

```diff
@@ -0,0 +1,205 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"testing"
+
+	pkgTest "github.com/knative/pkg/test"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	rtesting "github.com/knative/serving/pkg/reconciler/v1alpha1/testing"
+	"github.com/knative/serving/test"
+)
+
+func setServiceGenerateName(generateName string) rtesting.ServiceOption {
+	return func(service *v1alpha1.Service) {
+		service.ObjectMeta.GenerateName = generateName
+	}
+}
+
+func setConfigurationGenerateName(generateName string) rtesting.ConfigOption {
+	return func(config *v1alpha1.Configuration) {
+		config.ObjectMeta.GenerateName = generateName
+	}
+}
+
+func setRouteGenerateName(generateName string) rtesting.RouteOption {
+	return func(route *v1alpha1.Route) {
+		route.ObjectMeta.GenerateName = generateName
+	}
+}
+
+// generateNamePrefix returns the object name to be used for testing, shorted to
+// 44 characters to avoid #3236, as generateNames longer than 44 characters may cause
```

Comment:
The issue referenced seems to indicate 49 characters. I assume the text here then is expecting 5 additional from the generateName randomness?

Reply:
Yes, that is exactly it. 49 characters will work with a service with `.metadata.name` set but not `.metadata.generateName` set.